frontend/src/Frontend/UI/Widgets.hs

```diff
@@ -1108,6 +1110,36 @@ uiAccountNameInput label inlineLabel initval onSetName validateName = do
     & setValue .~ Just onSetName
   pure (tagPromptlyDyn v i, v)
 
+-- | Creates an input widget, along with an initial value and a setter event.
+-- The validation function will be used to filter out erroneous results.
+--
+-- Values that result in `Success` or `Warning` will be passed on. `Failure` values will be filtered out.
+--
+-- Values that result in `Failure` or `Warning` will be shown as a pop over error. `Success` values will not show errors.
+uiTextInputAsync
+  :: ( DomBuilder t m
+     , PostBuild t m
+     , MonadHold t m
+     , DomBuilderSpace m ~ GhcjsDomSpace
+     , PerformEvent t m
+     , TriggerEvent t m
+     , MonadJSM (Performable m)
+     , MonadFix m
+     )
+  => Text                     -- ^ The label for this input
+  -> Bool                     -- ^ Should the label be inline?
+  -> Maybe Text               -- ^ Initial Value for the input
+  -> Event t (Maybe Text)     -- ^ An event that can be used to set the input's value
+  -> Dynamic t (Event t Text -> m (Event t (ValidationResult Text a)))
```

Comment:
What is the point of this being Dynamic?

Reply:
Removed.